src/planet-a/PlanetGeometryA.cpp

```diff
@@ -38,21 +35,35 @@
 using namespace planeta;
 
 using osp::Vector3;
+using Magnum::Math::pow;
+using Magnum::Math::abs;
+
+void planeta::update_range_insert(std::vector<UpdateRangeSub>& range,
+                                  UpdateRangeSub insert)
+{
+    // TODO: Merge update ranges to decrease number of glBufferSubData calls
+
+    //UpdateRangeSub merged;
+
+    range.push_back(insert);
+}
 
-void PlanetGeometryA::initialize(float radius)
+void PlanetGeometryA::initialize(std::shared_ptr<IcoSphereTree> sphere,
+                                 unsigned chunkDiv, chindex_t maxChunks,
+                                 vrindex_t maxShared)
 {
     //m_subdivAreaThreshold = 0.02f;
-    m_vrtxSharedMax = 10000;
-    m_chunkMax = 100;
+    m_vrtxSharedMax = maxShared;
+    m_chunkMax = maxChunks;
 
     //m_chunkAreaThreshold = 0.04f;
-    m_chunkWidth = 17; // MUST BE (POWER OF 2) + 1. min: 5
-    m_chunkWidthB = m_chunkWidth - 1; // used often in many calculations
+    m_chunkWidthB = pow(2u, chunkDiv);
+    m_chunkWidth  = m_chunkWidthB + 1; // MUST BE (POWER OF 2) + 1. min: 5
```

Comment:
Then add an assert to guarantee it.

Reply:
This was previously set with magic numbers, the comment is redundant. The pow should guarantee it. Asserting if chunkDiv is too high is more reasonable.